kinesumer.go

```diff
@@ -488,30 +494,37 @@ func (k *Kinesumer) consumePipe(stream string, shard *Shard) {
 				}
 			}
 		case <-checkPointTicker.C:
-			k.commitCheckPoint(stream, shard.ID, lastSequence)
+			k.Commit(stream, shard.ID, lastSequence)
 		}
 	}
 }
 
-// update checkpoint using sequence number.
-func (k *Kinesumer) commitCheckPoint(stream, shardID, lastSeqNum string) {
-	if lastSeqNum == "" {
+// Commit updates check point with sequence number.
+func (k *Kinesumer) Commit(stream, shardID, sequenceNumber string) {
+	if sequenceNumber == "" {
 		// sequence number can't be empty.
+		k.errors <- ErrEmptySequenceNumber
 		return
 	}
 
 	ctx := context.Background()
 	ctx, cancel := context.WithTimeout(ctx, checkPointTimeout)
 	defer cancel()
 
-	if err := k.stateStore.UpdateCheckPoint(ctx, stream, shardID, lastSeqNum); err != nil {
-		log.Err(err).
-			Str("stream", stream).
-			Str("shard id", shardID).
-			Str("missed sequence number", lastSeqNum).
-			Msg("kinesumer: failed to UpdateCheckPoint")
+	if err := k.commitCheckPoint(ctx, stream, shardID, sequenceNumber); err != nil {
+		msg := fmt.Sprintf("kinesumer: failed to Commit %s/%s: %s", stream, shardID, sequenceNumber)
+		k.errors <- errors.Wrap(err, msg)
```

Comment:
errors pkg has "Wrapf" function. Use it instead. (and we don't need to add "kinesumer: " scope prefix the additional error messages)